cpp/oneapi/dal/algo/svm/backend/cpu/train_kernel_thunder.cpp

```diff
@@ -0,0 +1,116 @@
+
+template <typename Float>
+static train_result call_daal_kernel(const context_cpu& ctx,
+                                     const descriptor_base& desc,
+                                     const table& data,
+                                     const table& labels,
+                                     const table& weights) {
+    const std::int64_t row_count    = data.get_row_count();
+    const std::int64_t column_count = data.get_column_count();
+
+    // TODO: data is table, not a homogen_table. Think better about accessor - is it enough to have just a row_accessor?
+    auto arr_data    = row_accessor<const Float>{ data }.pull();
+    auto arr_label   = row_accessor<const Float>{ labels }.pull();
+    auto arr_weights = row_accessor<const Float>{ weights }.pull();
+
+    const auto daal_data =
+        interop::convert_to_daal_homogen_table(arr_data, row_count, column_count);
+    const auto daal_labels  = interop::convert_to_daal_homogen_table(arr_label, row_count, 1);
+    const auto daal_weights = interop::convert_to_daal_homogen_table(arr_weights, row_count, 1);
+
+    // TODO: move as parameter onedal SVM
+    auto kernel =
+        daal_kernel_function::KernelIfacePtr(new daal_kernel_function::linear::Batch<Float>());
+
+    daal_svm::Parameter daal_parameter(
+        kernel,
+        desc.get_c(),
+        desc.get_accuracy_threshold(),
+        desc.get_tau(),
+        desc.get_max_iteration_count(),
+        std::int64_t(desc.get_cache_size() * 1024 * 1024), // DAAL get in bytes
+        desc.get_shrinking());
+
+    auto daal_model = daal_svm::Model::create<Float>(column_count);
+    interop::call_daal_kernel<Float, daal_svm_thunder_kernel_t>(ctx,
+                                                                daal_data,
+                                                                daal_weights,
+                                                                *daal_labels,
+                                                                daal_model.get(),
+                                                                &daal_parameter);
+
+    auto table_support_vectors =
+        interop::convert_from_daal_homogen_table<Float>(daal_model->getSupportVectors());
+    auto table_support_indices =
+        interop::convert_from_daal_homogen_table<Float>(daal_model->getSupportIndices());
+    auto table_classification_coefficients = interop::convert_from_daal_homogen_table<Float>(
+        daal_model->getClassificationCoefficients());
+    const double bias                        = daal_model->getBias();
+    const std::int64_t support_vectors_count = table_support_vectors.get_row_count();
+
+    auto onedal_model = model()
+                            .set_support_vectors(table_support_vectors)
+                            .set_coefficients(table_classification_coefficients)
+                            .set_bias(bias)
+                            .set_support_vectors_count(support_vectors_count);
```

Comment:
I think it is better to extract it into a separate method like `interop::convert_from_daal_svm_model(daal_model)`

Reply:
Applied